### ISSUE TITLE
lesson_2 #15

### ISSUE THREAD

**Issue by jenessalemon** [pearselab/r-intro-jenessalemon#11]
@Wolflab @willpearse The code for this problem starts on line 348.
I have had this problem before. When I put a Boolean in to an if statement:

if (distance[i] >= 5){
    return(i)
    break
  }

I get an error saying:

Error in if (distance[i] >= 5) { : missing value where TRUE/FALSE needed

I thought that that must mean that distance is just full of NAs. However, I printed distance and got real values. Any ideas on how to fix this?

**Comment by willpearse** [comment]
Right, but did you print `distance[i]`? What does that look like? Try `print`ing that out in your loop and seeing what the value of `distance[i]` looks like just before you get the error...

**Comment by jenessalemon** [comment]
Crap it's NA. How do I fix this? It doesn't make sense for the whole thing to be fine but one iteration to be NA? I will just try printing at each step and see when it becomes NA...

**Comment by jenessalemon** [comment]
It looks like the values go up and up until the last iteration, where they become NA. So I tried using if (distance[i-1] >= 5)
and it worked...but that is probably not the best way to fix it!

**Comment by jenessalemon** [comment]
Actually that didn't fix it, it is telling me that the wanderer is never falling off the cliff, even though he is well past 5 mi away.

**Comment by willpearse** [comment]
Come online at https://appear.in/usubioprogramming
